src/connection_manager.c

```diff
@@ -332,7 +333,17 @@ static qd_error_t load_server_config(qd_dispatch_t *qd, qd_server_config_t *conf
     config->port                 = qd_entity_get_string(entity, "port");              CHECK();
     config->name                 = qd_entity_opt_string(entity, "name", 0);           CHECK();
     config->role                 = qd_entity_get_string(entity, "role");              CHECK();
-    config->inter_router_cost    = qd_entity_opt_long(entity, "cost", 1);             CHECK();
+    long inter_router_cost       = qd_entity_opt_long(entity, "cost", 1);             CHECK();
+
+    //
+    // The cost field on the listener or the connector should be > 0 and <= INT_MAX
+    // The router will terminate on invalid cost values.
+    //
+    if (inter_router_cost <= 0 || inter_router_cost > INT_MAX) {
```

Comment:
Hrm - shouldn't the constant be INT32_MAX from stdint.h?  limits.h says INT_MAX must be at least 2^32-1, which means some arch can define it to be larger.